### PR TITLE
Date format dart

### LINKED ISSUE #2696: [BUG] Dart code generator does not support date format for string
##### Description

The dart code generator does support date-time format for strings, making the parameter type DateTime and formatting it correctly. However if the format for a string is "date" it also uses DateTime for the parameter, but it does not format it correctly as it includes the time in the formatted version.

##### openapi-generator version

4.0.0-beta3

### PR DESCRIPTION
**Pull request by sbu-WBT**
Fixes #2696

If something is missing or other changes have to be done, I'm open for any suggestions.

I'm not sure if the petstore changes are correct. I executed all commands from dart-petstore.bat

@ircecho @swipesight @jaumard @athornz @amondnet

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by sbu-WBT** [comment]
Guys, I don't know why the CI builds fail. I re-updated the samples with the new instructions. I used the git bash on windows to run `generate-samples.sh`. It seams, that on windows backslashes and else forward slashes are used. I really need this fix so I would appreciate any help. @wing328

**Comment by agilob** [comment]
LGTM but would like to see a test for it `samples/client/petstore/dart2/petstore/test/store_faked_client_test.dart`

**Comment by wing328** [comment]
I'll take a look in the coming week.

File: modules/openapi-generator/src/main/resources/dart2/api_helper.mustache [hunk 1]
```diff
@@ -1,6 +1,7 @@
 part of {{pubName}}.api;
 
 const _delimiters = const {'csv': ',', 'ssv': ' ', 'tsv': '\t', 'pipes': '|'};
+final _dateFormatter = DateFormat('yyyy-MM-dd');
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/dart2/api_helper.mustache:4]: Am I understanding it correctly that _dateFormatter cannot be changed to other date formats during runtime?

Comment (sbu-WBT) [reply modules/openapi-generator/src/main/resources/dart2/api_helper.mustache:4]: My thoughts were to protect it against accidentally change. I don't see any reasons to allow modification as the specification says it hast to be this format https://swagger.io/specification/#dataTypes. But if you wish, I can change it

Comment (wing328) [reply modules/openapi-generator/src/main/resources/dart2/api_helper.mustache:4]: Ideally, all REST API should be using the date format following RFC3339 according to OpenAPI spec but there were use cases before in which date using a different format and that's why we make it configurable, e.g. C# configuration class: https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/csharp/OpenAPIClient/src/Org.OpenAPITools/Client/Configuration.cs#L100

Comment (sbu-WBT) [reply modules/openapi-generator/src/main/resources/dart2/api_helper.mustache:4]: Okay, I will remove the final

**Comment by sbu-WBT** [comment]
Everything is fine now, isn't it? @wing328

**Comment by wing328** [comment]
@sbu-WBT PR merged. Thanks for your contribution.

**Pull request merged**